### ISSUE TITLE
bug: [React] IonModal is not removed from DOM when parent component unmounts

### ISSUE THREAD

**Issue by puopg** [ionic-team/ionic-framework#26185]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When a component rendering an IonModal is unmounted by React, the ion modal DOM node is not removed.

### Expected Behavior

Expected similar/same behavior as Ionic v5, where when a React component unmounts, the corresponding components being rendered by that component are also cleaned up. Even if there is no animation, I believe the DOM should be cleaned up

### Steps to Reproduce

1. Have a component (Call it `MyComponent`) that renders an IonModal
2. Conditionally mount that component somewhere.
3. Toggle the modal to present by changing `isOpen` to true.
4. Now, without toggling the modal's `isOpen` prop, simply toggle the mounting condition so React unmounts `MyComponent`.
5. The Modal is not removed from the DOM

### Code Reproduction URL

https://codesandbox.io/s/ionic-6-3-3-modal-unmount-bug-0etb2k?file=/src/pages/Home.tsx
_No response_

### Ionic Info

Ionic React: 6.3.3

### Additional Information

Link to codesandox example:  https://codesandbox.io/s/ionic-6-3-3-modal-unmount-bug-0etb2k?file=/src/pages/Home.tsx

**Comment by sean-perkins** [comment]
@puopg thanks for reporting this issue! 

I believe this may overlap with #25775, but we can track independently in the mean time. 

Now that #25590 has been resolved, I can port over the exploration fix for dismissing any presented overlay when it's react component is unmounted.

**Comment by puopg** [comment]
Ah I see, for some reason I thought https://github.com/ionic-team/ionic-framework/issues/25775 was resolved with your fix and just not marked! Thanks for looking into this!

**Comment by sean-perkins** [comment]
Hello @puopg can you test with this dev-build and let me know if you run into any issues?

```bash
npm install @ionic/react@6.3.5-dev.11667931176.1678d34e @ionic/react-router@6.3.5-dev.11667931176.1678d34e
```

I've additionally forked your reproduction here: https://codesandbox.io/s/ionic-6-3-3-modal-unmount-bug-forked-dflgj8

**Comment by BenHen75** [comment]
I had the same issue, your fix work perfectly @sean-perkins thx!!

**Comment by puopg** [comment]
Hey, sorry i haven't been around to get to this. Ill check this out when I can and update!

**Comment by puopg** [comment]
@sean-perkins Hey so far I think things looking good!

**Comment by puopg** [comment]
@filip-dot-cube The PR is still open, so its not available on an official release to my understanding yet.

https://github.com/ionic-team/ionic-framework/pull/26245

**Comment by sean-perkins** [comment]
The PR to resolve this issue has been merged 🎉 and will be available in the next patch release of Ionic v6.6.1.

We appreciate the community's patience with this issue!
